test/find_pattern_test.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// find_pattern_test.dart
+// Tests for Find Pattern
+//
+// 2025 March 6
+// Author: Louiz Ang Zhi Lin <louiz.zhi.lin.ang@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
suggestion: try writing at least one test where you do multiple different values on the bus, pattern, and `n` for the same hardware.  this will help test against the type of `.toInt()` bugs that you have in your implementation

Reply:
> suggestion: try writing at least one test where you do multiple different values on the bus, pattern, and `n` for the same hardware. this will help test against the type of `.toInt()` bugs that you have in your implementation

Can you describe more what do you mean by multiple values for the same hardware here?

Reply:
Yeah, so the hardware can exist with input signals changing, and the outputs would dynamically change.

For example, a test could look something like this:

```dart
final bus = Logic(width:8);
final pattern = Logic(width:2);

final findPattern = FindPattern(bus, pattern);

bus.put(bin('10000001'));
pattern.put(bin('01'));
expect(findPattern.index.value.toInt(), equals(0));

bus.put(bin('10001001'));
pattern.put(bin('10'));
expect(findPattern.index.value.toInt(), equals(2));
```

this way, if you accidentally have some generation-time calculations done, the hardware outputs wont dynamically update when the inputs changs, so it's a good test for the types of issues you had in your implementation previously